### PR TITLE
base1: Keep channel open on unknown messages

### PR DESCRIPTION
**Pull request by petervo**
Future compatibility for #2424 type issues.

### CODE REVIEW

File: pkg/base1/cockpit.js [hunk 1]
```diff
@@ -1632,7 +1632,6 @@ function full_scope(cockpit, $, po) {
                 owner = msg.owner;
             } else {
                 console.warn("received unexpected dbus json message:", payload);
```
Comment (stefwalter) [inline pkg/base1/cockpit.js]: This isn't an error. Perhaps worth a debug line, but not a ```console.warn```. Isn't it expected that new messages will be added to ```dbus-json3```? That's the intent of this change, right?

**Comment by petervo** [comment]
Updated this PR to only send the owned signal when explicitly asked for. Other options for handling this are a more generic, "can_ignore" option or a json-dbus version bump.

File: src/bridge/cockpitdbusjson.c [hunk 2]
```diff
@@ -1932,6 +1935,12 @@ cockpit_dbus_json_prepare (CockpitChannel *channel)
       goto out;
     }
 
+  if (!cockpit_json_get_bool (options, "notify_owner", FALSE, &self->notify_owner))
```
Comment (mvollmer) [inline src/bridge/cockpitdbusjson.c]: This needs to be documented.

**Comment by petervo** [comment]
Added documentation in protocol.md

**Comment by stefwalter** [comment]
We're going to remove the 'notify_owner' stuff from this pull request right? And just fix the bug?

**Comment by petervo** [comment]
removed notify_owner.

**Comment by stefwalter** [comment]
Should we add a test?

**Comment by petervo** [comment]
Added a test. Figured that using frames to inject an unknown message was better than trying to adapt test-server to support this sort of thing.

**Pull request closed without merging**